crates/pallets/starknet/src/lib.rs

```diff
@@ -1134,11 +1134,10 @@ impl<T: Config> Pallet<T> {
     pub fn simulate_transactions(
         transactions: Vec<UserTransaction>,
         simulation_flags: SimulationFlags,
-    ) -> Result<Vec<Result<TransactionExecutionInfo, PlaceHolderErrorTypeForFailedStarknetExecution>>, DispatchError>
-    {
+    ) -> Result<Vec<(TransactionSimulationResult, CommitmentStateDiff)>, DispatchError> {
         let chain_id = Self::chain_id();
 
-        let tx_execution_results = execute_txs_and_rollback::<T>(
+        let tx_execution_results = execute_txs_and_rollback_with_state_diff::<T>(
```

Comment:
When you will rebase the method won't be there. it's moved to pallet/starnet/src/simulations.rs